File: mobile/modules/bluetooth-sdk/ios/Source/Requests/DisplayRequests.swift
import Foundation

public struct DisplayTextRequest {
    public let text: String
    public let x: Int
    public let y: Int
    public let size: Int

    public init(text: String, x: Int = 0, y: Int = 0, size: Int = 24) {
        self.text = text
        self.x = x
        self.y = y
        self.size = size
    }

    var dictionary: [String: Any] {
        [
            "text": text,
            "x": x,
            "y": y,
            "size": size,
        ]
    }
}

struct DisplayEventRequest {
    let values: [String: Any]

    init(values: [String: Any]) {
        self.values = values
    }
}

public struct DashboardPositionRequest {
    public let height: Int
    public let depth: Int

    public init(height: Int, depth: Int) {
        self.height = height
        self.depth = depth
    }
}

struct DashboardMenuItem {
    let title: String
    let packageName: String
    let values: [String: Any]

    init(title: String, packageName: String, values: [String: Any] = [:]) {
        self.title = title
        self.packageName = packageName
        self.values = values
    }

    var dictionary: [String: Any] {
        values.merging(["title": title, "packageName": packageName]) { _, new in new }
    }
}

File: mobile/modules/bluetooth-sdk/android/src/main/java/com/mentra/bluetoothsdk/requests/DisplayRequests.kt
package com.mentra.bluetoothsdk

data class DisplayTextRequest(
    val text: String,
    val x: Int = 0,
    val y: Int = 0,
    val size: Int = 24,
) {
    fun toMap(): Map<String, Any> =
        mapOf(
            "text" to text,
            "x" to x,
            "y" to y,
            "size" to size,
        )
}

internal data class DisplayEventRequest(
    val values: Map<String, Any>,
) {
    fun toMap(): Map<String, Any> = values
}

data class DashboardPositionRequest(
    val height: Int,
    val depth: Int,
)

internal data class DashboardMenuItem(
    val title: String,
    val packageName: String,
    val values: Map<String, Any> = emptyMap(),
) {
    fun toMap(): Map<String, Any> =
        values + mapOf(
            "title" to title,
            "packageName" to packageName,
        )
}
